dsgrid/config/project_config.py

```diff
@@ -289,6 +306,143 @@ def dimension_mapping_manager(self):
     def dimension_mapping_manager(self, val: DimensionMappingRegistryManager):
         self._dimension_mapping_mgr = val
 
+    def get_base_dimension(self, dimension: DimensionType):
+        """Return the base dimension matching dimension_type.
+
+        Parameters
+        ----------
+        dimension_type : DimensionType
+
+        Returns
+        -------
+        DimensionConfig
+
+        """
+        for key, dim_config in self.base_dimensions.items():
+            if key.type == dimension:
+                return dim_config
+        assert False, dimension
+
+    def get_supplemental_dimensions(self, dimension: DimensionType):
+        """Return the supplemental dimensions matching dimension (if any).
+
+        Parameters
+        ----------
+        dimension_type : DimensionType
+
+        Returns
+        -------
+        DimensionConfig
+
+        """
+        return [v for k, v in self.supplemental_dimensions.items() if k.type == dimension]
+
+    def get_base_to_base_dimension_mapping_by_types(
+        self, from_dimension: DimensionType, to_dimension: DimensionType
+    ):
+        """Return the base-to-base dimension mapping matching the dimension types.
+
+        Parameters
+        ----------
+        from_dimension : DimensionType
+        to_dimension : DimensionType
+
+        Returns
+        -------
+        DimensionMappingConfig
+
+        """
+        for config in self._base_to_base_mappings.values():
+            if (
+                config.model.from_dimension.dimension_type == from_dimension
+                and config.model.to_dimension.dimension_type == to_dimension
+            ):
+                return config
+        raise DSGInvalidDimensionMapping(
+            f"mappings are not stored: from={from_dimension} to={to_dimension}"
+        )
+
+    def get_base_to_supplemental_dimension_mappings_by_types(self, dimension: DimensionType):
+        """Return the base-to-supplemental dimension mappings for the dimension (if any).
+
+        Parameters
+        ----------
+        dimension : DimensionType
+
+        Returns
+        -------
+        list
+            List of DimensionMappingConfig
+
+        """
+        return [
+            x
+            for x in self._base_to_supplemental_mappings.values()
+            if x.model.from_dimension.dimension_type == dimension
+        ]
+
+    def has_base_to_base_dimension_mapping_types(self, from_dimension, to_dimension):
+        """Return True if the config has these base-to-base mappings."""
+        return self._has_mapping(from_dimension, to_dimension, self._base_to_base_mappings)
```

Comment:
Flagging another instance of base-to-base

Reply:
Looking farther down, we do of course have dataset-base to project-base mappings. Then I think we're calling within project base-to-base mappings "association mappings" instead. Is that correct? 

Without looking a lot closer I can't tell which kind of mapping (of the two I just listed) is meant here.

Reply:
All references to base-to-base mappings will be deleted in the next push to this PR.